app/src/main/java/com/example/android/codelabs/paging/data/GithubRepository.kt

```diff
@@ -17,19 +17,21 @@
 package com.example.android.codelabs.paging.data
 
 import android.util.Log
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
 import com.example.android.codelabs.paging.api.GithubService
 import com.example.android.codelabs.paging.api.searchRepos
-import com.example.android.codelabs.paging.db.GithubLocalCache
+import com.example.android.codelabs.paging.model.Repo
 import com.example.android.codelabs.paging.model.RepoSearchResult
 
 /**
  * Repository class that works with local and remote data sources.
  */
-class GithubRepository(
-    private val service: GithubService,
-    private val cache: GithubLocalCache
-) {
+class GithubRepository(private val service: GithubService) {
+
+    // keep the list of responses
+    private val inMemoryCache = MutableLiveData<List<Repo>>()
 
     // keep the last requested page. When the request is successful, increment the page number.
     private var lastRequestedPage = 1
```

Comment:
You should be able to get rid of the `networkErrors` and `isRequestInProgress` tracking since we'll have callbacks for load state updates and retry built into paging 3. But maybe the point is to show that this logic would be replaced when you put paging3 in?

Reply:
Yep - this will be removed when moving to paging